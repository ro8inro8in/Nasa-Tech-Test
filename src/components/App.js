import React, { useState } from "react";
import "../styles/App.css";
import Search from "./Search";
import SearchResults from "./SearchResults";

function App() {
  const [searchResults, setSearchResults] = useState();
  return (
    <div className="App">
      <img
        className="nasa-logo"
        src="https://cdn.cnn.com/cnnnext/dam/assets/200424060716-nasa-worm-logo.jpg"
        alt="nasa-logo"
      />
      <Search setSearchResults={setSearchResults} />
      <div className="search-results">
        <SearchResults results={searchResults} />
      </div>
    </div>
  );
}

export default App;
