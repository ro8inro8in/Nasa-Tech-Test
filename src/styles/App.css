.App {
    display: flex;
    flex-direction: column;
  }
  
  .nasa-logo {
    margin: auto;
    width: 500px;
    height: 300px;
  }

