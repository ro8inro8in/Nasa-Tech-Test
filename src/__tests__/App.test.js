import React from "react";
import { render, screen } from "@testing-library/react";
import App from "../components/App";

describe("App", () => {
  const { asFragment } = render(<App />);

  it("renders correctly", () => {
    expect(asFragment()).toMatchSnapshot();
  });
});
