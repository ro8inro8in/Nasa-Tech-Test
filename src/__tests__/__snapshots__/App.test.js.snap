// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <img
      alt="nasalogo"
      class="nasa-logo"
      src="https://cdn.cnn.com/cnnnext/dam/assets/200424060716-nasa-worm-logo.jpg"
    />
    <div
      class="search"
    >
      <form
        class="search-form"
      >
        <input
          class="search-input"
          type="text"
        />
        <button
          class="search-btn"
          type="submit"
        >
          Go!
        </button>
      </form>
    </div>
    <p>
      Search Results
    </p>
    <img
      alt="planet-image"
      class="planet-image"
      src="https://images.unsplash.com/photo-1522030299830-16b8d3d049fe?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&w=1000&q=80"
    />
  </div>
</DocumentFragment>
`;
